### PR TITLE
unit test controller

### PR DESCRIPTION
**Pull request by phuongha47**
## Related Tickets:
- [#45978](https://edu-redmine.sun-asterisk.vn/issues/45978)

## WHAT this PR does?
-  Unit test UserController
-  Fix repository

## Notes Impacted Areas:
- N/A.

## Attachments:
- N/A.

### CODE REVIEW

**Comment by phuongha47** [comment]
ready

**Comment by Khoacannotcode** [comment]
Okay nice

**Comment by phuongha47** [comment]
ready

File: tests/Unit/Controllers/UserControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    {
+        Hash::shouldReceive('make')
+            ->once()
+            ->andReturn('hashed');
+        $user = new User(
+            [
+            'name' => 'User',
+            'password' => 'rawpassword',
+            ]
+        );
```
Comment (ghost) [inline tests/Unit/Controllers/UserControllerTest.php]:
```suggestion
        $user = new User([
            'name' => 'User',
            'password' => 'rawpassword',
        ]);
```

File: tests/Unit/Controllers/UserControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    //  test_update
+    public function testUpdateUser()
+    {
+        $user = User::factory()->make();
+        $user_edit_info = new UserEditRequest(
+            [
+            'name' => 'jmac',
+            'email' => 'jmac@example.com',
+            ]
+        );
```
Comment (ghost) [inline tests/Unit/Controllers/UserControllerTest.php]:
```suggestion
        $user_edit_info = new UserEditRequest([
            'name' => 'jmac',
            'email' => 'jmac@example.com',
        ]);
```

File: tests/Unit/Controllers/UserControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    {
+        $user = new UserAddRequest(
+            [
+            'name' => 'jmac',
+            'email' => 'jmac@example.com',
+            'password' => 'password',
+            'password_confirmation' => 'password',
+            'role_id' => 1,
+            ]
+        );
```
Comment (ghost) [inline tests/Unit/Controllers/UserControllerTest.php]:
```suggestion
        $user = new UserAddRequest([
            'name' => 'jmac',
            'email' => 'jmac@example.com',
            'password' => 'password',
            'password_confirmation' => 'password',
            'role_id' => 1,
        ]);
```

File: app/Repositories/BaseRepository.php [hunk 4]
```diff
@@ -44,7 +44,7 @@ public function create($attributes = [])
     {
         return $this->model->create($attributes);
     }
-
+
```
Comment (ghost) [inline app/Repositories/BaseRepository.php]: xóa khoảng trắng thừa

**Comment by phuongha47** [comment]
ready

**Pull request merged**